catalog/harvest/catalog-harvest-file/src/main/java/org/codice/ddf/catalog/harvest/file/DirectoryHarvester.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest.file;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.monitor.FileAlterationListenerAdaptor;
+import org.apache.commons.io.monitor.FileAlterationObserver;
+import org.codice.ddf.catalog.harvest.Listener;
+import org.codice.ddf.catalog.harvest.common.FileSystemPersistenceProvider;
+import org.codice.ddf.catalog.harvest.common.PollingHarvester;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DirectoryHarvester extends PollingHarvester {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DirectoryHarvester.class);
+
+  private static final long DEFAULT_POLL_INTERVAL = 5;
+
+  private final Set<Listener> listeners = new HashSet<>();
```

Comment:
❗️This `Set` needs to be synchronized to prevent possible threading issues when listeners are being added or removed while notifications are being sent by the `DirectoryHarvesterListenerAdaptor`. Recommend taking a look at https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/event/EventListenerSupport.html.